contrib/kube-prometheus/manifests/grafana/grafana-deployment.yaml

```diff
@@ -27,6 +27,8 @@ spec:
             secretKeyRef:
               name: grafana-credentials
               key: password
+        - name: GF_SERVER_ROOT_URL
```

Comment:
I understand your need for this, but I think this might be too opinionated, what if people prefer accessing grafana via ingress, or node port?

Reply:
I totally understand your argument so far we've tried to be as un-opinionated as possible (although not totally unavoidable) about the namespace things are deployed in, and Prometheus/Alertmanager currently also use node port services as examples. Something to note about the kube-prometheus manifest collection is that it's just that: a collection of manifests to start out with, it's totally meant to take and adapt :slightly_smiling_face: . The helm charts (which do admittedly need a lot of work) on the other hand are totally meant to be configurable in terms of nodeport/proxy/port-forward/ingress as the method of access. Does that make sense?

Reply:
@brancz OK I see what you mean about other services in kube-prometheus also being node ports by default. However, are you sure it doesn't make sense to align with the official Elasticsearch/Fluentd/Kibana add-on in this regard? They want to be unopinionated as well, but as you have to take a certain stand with the default configuration, they went with supporting `kubectl proxy` as the default. 

Regardless though, does it make sense to name the service port ("web")? It was extremely tricky to figure out that the port name was the reason that the typical `kubectl proxy` URL to the Grafana service didn't work (requiring a `:web` suffix).

Reply:
The thing about proxying through the apiserver is that it exactly requires this kind of odd URL prefixing and the application to know it's external accessible URL, which in turn means that we need to put this information in all of the manifests we provide which in turn hardcodes which namespace they run in, and the name of the service they are exposed by. By a user it's not obvious that if one of these parameters are changed that this opaque string needs to be changed. Aside from that it's easier to comprehend for users to add something to a manifests, than to remove something. I realize that proxying is not a bad default, but certainly not a production solution either. Once the required components are more matured (mainly the bitly oauth2 proxy) I'd like to provide samples for ingress instead of node port as I believe that's the closest to what people actually want.

Regarding your other question: I think the service port naming makes sense the way you proposed.